source/Symbol/SwiftASTContext.cpp

```diff
@@ -3551,7 +3551,11 @@ void SwiftASTContext::LoadModule(swift::ModuleDecl *swift_module,
         all_dlopen_errors.GetData());
   };
 
-  swift_module->collectLinkLibraries(addLinkLibrary);
+  swift_module->forAllVisibleModules(
+      {}, [&](swift::ModuleDecl::ImportedModule import) {
+        import.second->collectLinkLibraries(addLinkLibrary);
+        return true;
+      });
```

Comment:
Ahh…the correct fix is to call `collectLinkLibraries` on the SourceFile being compiled, not the module. Sorry for our mistake!